added a return_indices option in cross_validation::train_test_split()

rbahumi opened the pull request:
Added a boolean return_indices option to cross_validation.py's train_test_split() method
  If True, the indices of the train and test split will additionally be returned.

Motivation: I needed the indices of the cross validation split, and I saw online that I wasn't the only one having this use case:
https://twitter.com/treycausey/status/423193367745286144

Please note - This is my first push to this repository. I did not run any tests to validate this code because I didn't know where the tests are located.

rvraghav93 commented:
The tests are to be added to `scikit-learn/sklearn/tests/test_cross_validation.py`

@vene @jnothman @amueller please take a look :)

rvraghav93 commented on sklearn/cross_validation.py:
```diff
@@ -1577,7 +1585,9 @@ def train_test_split(*arrays, **options):
-        List containing train-test split of input array.
+        List containing train-test split of input array. If return_indices
+        is True, the indices of the train and test split will additionally be
+        returned. length=2 * (len(arrays) + 1)
```
lenght=2*... should be removed :)

rbahumi replied:
Removed :-)

rvraghav93 commented:
Thanks for addressing the comment! Please do add a test :)

rbahumi committed:
return_indices option in cross_validation::train_test_split():
  Added a boolean return_indices option to cross_validation.py's train_test_split() method
  If True, the indices of the train and test split will additionally be returned.

rbahumi commented on sklearn/tests/test_cross_validation.py:
```diff
@@ -710,6 +710,27 @@ def test_train_test_split():
         assert_equal(np.sum(train == 1), np.sum(train == 2))
 
 
+def test_train_test_split_with_indices():
```
@rvraghav93, test was added. Is this test satisfying?

jnothman commented:
Perhaps this is a useful helper for some, but it can be achieved simply by invoking:
```python
train_test_split(X, y, np.arange(X.shape[0]), ...)
```
Is there a way we can make that clearer to users without complicating the interface?

amueller commented:
I think we shouldn't add it as it is so readily implemented as @jnothman points out. We can add it to the docstring example maybe?

amueller commented:
closing for lack of enthusiasm
